### DISCUSSION TITLE
how do I force xmake to build in debug mode by default?

### DISCUSSION THREAD

**Discussion by smac89** [xmake-io/xmake#4864] [Q&A] [answered]
I have set the following in my config:

```lua
option("mode")
    set_default("debug")
```

Then I run:

```
xmake f -c
xmake
```

But it always default to building in release mode.

I even run `xmake f -h`, and the last few lines of the output shows:

```
Command options (Project Configuration):

        --mode=MODE                        The mode option (default: debug)
```

So clearly, `xmake f -c` is not properly picking up that it should generate a debug config.

**Comment by waruqi** [comment, marked as answer]
`set_defaultmode("debug")`

**Reply by smac89** [reply]
I appreciate it. Thanks

**Reply by smac89** [reply]
I also think I may have stumbled on a bug #4865

**Reply by yh-sb** [reply]
@waruqi, Is it the same as `set_config("mode", "debug")` ?

**Reply by waruqi** [reply]
right, but set_defaultmode will be better.
